src/components/specific/tags/tags-main/TagsMain.vue

```diff
@@ -0,0 +1,211 @@
+      </template>
+      <div class="tags-main__content__search-bar">
+        <BIMDataSearch
+          v-model="filter"
+          :placeholder="$t('Tag.search')"
+          color="primary"
+          radius
+          width="100%"
+          height="calc(var(--spacing-unit) * 2.5)"
+        />
+      </div>
+      <div class="tags-main__content__tag-list">
+        <template v-for="tag of updatedTagList" :key="tag.id">
+          <TagsItem
+            v-if="tag.isFindable"
+            :project="project"
+            :document="document"
+            :tagsMain="tagsMain"
+            :tag="tag"
+            @tag-to-update="tagToUpdate = $event"
+            @fetch-tags="$emit('fetch-tags')"
+            @file-updated="$emit('file-updated')"
+          />
+        </template>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { ref, watch, onMounted } from "vue";
+import { getRandomHexColor } from "@/utils/colors.js";
+import TagsItem from "@/components/specific/tags/tags-item/TagsItem.vue";
+import { useToggle } from "@/composables/toggle";
+import TagService from "@/services/TagService";
+import { useFiles } from "@/state/files.js";
+
+export default {
+  components: { TagsItem },
+
+  props: {
+    project: {
+      type: Object,
+      required: true
+    },
+    document: {
+      type: Object,
+      required: true
+    },
+    allTags: {
+      type: Array,
+      default: () => []
+    }
+  },
+  emits: ["close", "fetch-tags", "file-updated"],
+  setup(props, { emit }) {
+    const { getDocument } = useFiles();
+    const { isOpen: showAddTagInput, toggle: toggleAddTagInput } = useToggle();
+
+    const tagsDocument = ref(null);
+    const tagsMain = ref(null);
+
+    const input = ref(null);
+
+    watch(showAddTagInput, () =>
+      setTimeout(() => showAddTagInput.value && input.value.focus(), 100)
+    );
+
+    const newTagName = ref("");
+
+    const addNewTag = async () => {
+      if (newTagName.value) {
+        await TagService.createTag(props.project, {
+          name: newTagName.value,
+          color: getRandomHexColor()
+        });
+        newTagName.value = "";
+        toggleAddTagInput();
+        emit("file-updated");
+        emit("fetch-tags");
+      }
+    };
+
+    const updatedTagList = ref(null);
+    const filter = ref("");
+
+    const getTagListUpdated = (list = updatedTagList.value) => {
+      updatedTagList.value = list
+        .map(tag => {
+          return {
+            ...tag,
+            isFindable:
+              filter.value === ""
+                ? true
+                : tag.name.includes(filter.value.toLowerCase()),
+            isSelected: tagsDocument?.value.some(t => t.id === tag.id)
+          };
+        })
+        .sort((a, b) => (a.id > b.id ? 1 : -1));
+    };
+
+    const tagToUpdate = ref(null);
+
+    watch(tagToUpdate, async () => {
+      if (!tagToUpdate.value) return;
+      updatedTagList.value.map(tag =>
+        tag.id === tagToUpdate.value.id
```

Comment:
What's the purpose of this mapping ?
It looks like that you do not assign the value of this mapping, so I suspect that this has no effect at all.
It may not be necessary in the end.